pydantic/_internal/_dataclasses.py

```diff
@@ -62,6 +62,8 @@ def prepare_dataclass(
             'Support for `__post_init_post_parse__` has been dropped, the method will not be called', DeprecationWarning
         )
 
+    cls = cast('type[PydanticDataclass]', cls)
```

Comment:
Yeah you get that because that attribute hasn['t been set. We're literally setting it a couple lines below. But without that cast pyright complains that you're assigning an unknown attribute.